prov/util/src/util_attr.c

```diff
@@ -1014,8 +1015,6 @@ static void fi_alter_domain_attr(struct fi_domain_attr *attr,
 	if (FI_VERSION_LT(api_version, FI_VERSION(1, 5)))
 		attr->mr_mode = attr->mr_mode & OFI_MR_BASIC_MAP ?
 			FI_MR_BASIC : FI_MR_SCALABLE;
-	else
-		attr->mr_mode &= ~(FI_MR_BASIC | FI_MR_SCALABLE);
```

Comment:
This change in its current form breaks verbs provider. Assuming the app shouldn't get an fi_info with FI_MR_BASIC or FI_MR_SCALABLE bits set if it hasn't set them in the hints, we should add a check for the same here and keep the code that turns off the bits.

The else case should add a condition:
`else if (!(hints->mr_mode & (FI_MR_BASIC | FI_MR_SCALABLE)))`

Reply:
I don't know that we should be altering those bits at all. The above coercion is necessary for translation to older applications, but newer applications can explicitly request basic registration. If the check function fails, then the bits are incorrect to begin with. If the provider is returning bits to an application that is the wrong version, then that is a provider problem. ofi_alter_info is supposed to alter information structures for the purpose of returning version specific information back to applications from version specific providers. 

Somewhere, it needs to check the following:

```C
if ((mr_mode & (FI_MR_BASIC | FI_MR_SCALABLE)) && 
     (mr_mode & ~(FI_MR_BASIC | FI_MR_SCALABLE))
    return -FI_EINVAL; /* new mode bits cannot be used with old mode bits */
```

but, that isn't with the provided code, it is a problem with the utility check attributes function. Check attributes should find an issue long before ofi_alter_info is a factor.

Reply:
@a-ilango Help me get on the same page with test results. What does your command line look like when you run fabtests?

Reply:
I think we need to look at removing the OFI_CHECK_MR_BASIC macro.  It's not intuitive.

Is the util code passing through the API version to the underlying provider?

Reply:
> I think we need to look at removing the OFI_CHECK_MR_BASIC macro. It's not intuitive.

It can be replaced with a check for FI_MR_BASIC mode bit.

> Is the util code passing through the API version to the underlying provider?

Are  you asking about ofix_getinfo? That passes the version info down.